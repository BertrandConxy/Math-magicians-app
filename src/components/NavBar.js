import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quotes',
      text: 'Quotes',
    },
  ];
  return (
    <header>
      <h1>Math Magicians</h1>
      <nav>
        <ul>
          {links.map((link) => (
            <li key={link.id}>
              <NavLink to={link.path}>{link.text}</NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default NavBar;
