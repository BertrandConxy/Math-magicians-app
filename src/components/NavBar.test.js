import React from 'react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavBar from './NavBar';

describe('Navbar renders properly', () => {
  it('renders properly', () => {
    const component = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it(' adds the active class to the clicked link', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    const list = screen.getByRole('list').childElementCount;
    expect(list).toEqual(3);
    const link = screen.getByText('Calculator');
    const otherLink = screen.getByText(/home/i);
    userEvent.click(link);
    expect(link.closest('a')).toHaveAttribute('class', 'active');
    expect(otherLink.closest('a')).not.toHaveAttribute('class', 'active');
  });
});
