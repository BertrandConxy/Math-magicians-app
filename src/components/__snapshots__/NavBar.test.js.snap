// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders properly renders properly 1`] = `
<header>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
